enumerables.rb

```diff
@@ -0,0 +1,139 @@
+module Enumerable
+  # 1. Create my_each
+  def my_each
+    (0..length - 1).each do |i|
+      yield self[i]
+    end
+  end
+
+  # 2. Create my_each_with_index
+  def my_each_with_index
+    (0..length - 1).each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  # 3. Create my_select
+  def my_select
+    arr = []
+    (0..length - 1).each do |i|
+      criteria = yield(self[i])
+      arr << self[i] if criteria
+    end
+    arr
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`select`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_select.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] returns an enumerator if no block is given
```ruby
array.my_select == Enumerator #true
```